server/src/test/java/com/broadcom/lsp/cobol/service/delegates/references/SemanticElementOccurrencesTest.java

```diff
@@ -0,0 +1,174 @@
+    verify(SemanticLocations_2).containsToken(DOCUMENT, TOKEN);
+  }
+
+  @Test
+  void whenFindReferenceAfterAnalysis_thenReturnReferenceList() {
+    TextDocumentPositionParams position = getTextDocumentPositionParams();
+    getAfterAnalysisMocks();
+    List<Location> references =
+        occurrences.findReferences(DOCUMENT, position, new ReferenceContext());
+    assertEquals(1, references.size());
+    verify(semanticLocations_1).containsToken(DOCUMENT, TOKEN);
+    verify(SemanticLocations_2).containsToken(DOCUMENT, TOKEN);
+  }
+
+  @Test
+  void whenFindReferenceAfterAnalysisWithReferenceContext_thenReturnReferenceList() {
+    TextDocumentPositionParams position = getTextDocumentPositionParams();
+    getAfterAnalysisMocks();
+    List<Location> references =
+        occurrences.findReferences(DOCUMENT, position, new ReferenceContext(true));
+    assertEquals(2, references.size());
+    verify(semanticLocations_1, times(2)).containsToken(DOCUMENT, TOKEN);
+    verify(SemanticLocations_2, times(2)).containsToken(DOCUMENT, TOKEN);
+  }
+
+  @Test
+  void whenFindHighlightsBeforeAnalysis_thenReturnEmptyList() {
+    TextDocumentPositionParams position = getTextDocumentPositionParams();
+    getMocksForBeforeAnalysis();
+    List<DocumentHighlight> highlights = occurrences.findHighlights(DOCUMENT, position);
+    assertEquals(0, highlights.size());
+    verify(semanticLocations_1, times(2)).containsToken(DOCUMENT, TOKEN);
+    verify(SemanticLocations_2, times(2)).containsToken(DOCUMENT, TOKEN);
+  }
+
+  @Test
+  void whenFindHighlightsAfterAnalysis_thenReturnList() {
+    TextDocumentPositionParams position = getTextDocumentPositionParams();
+    getAfterAnalysisMocks();
+    List<DocumentHighlight> highlights = occurrences.findHighlights(DOCUMENT, position);
+    assertEquals(2, highlights.size());
+    verify(semanticLocations_1, times(2)).containsToken(DOCUMENT, TOKEN);
+    verify(SemanticLocations_2, times(2)).containsToken(DOCUMENT, TOKEN);
+  }
+
+  private TextDocumentPositionParams getTextDocumentPositionParams() {
+    TextDocumentPositionParams position = new TextDocumentPositionParams();
+    TextDocumentIdentifier textDocument = new TextDocumentIdentifier(URI);
+    position.setTextDocument(textDocument);
+    position.setPosition(new Position(0, 0));
+    return position;
+  }
+
+  private void getMocksForBeforeAnalysis() {
+    when(semanticLocations_1.containsToken(any(CobolDocumentModel.class), anyString()))
+        .thenReturn(false);
+    when(SemanticLocations_2.containsToken(any(CobolDocumentModel.class), anyString()))
+        .thenReturn(false);
+  }
+
+  private void getAfterAnalysisMocks() {
+    when(semanticLocations_1.containsToken(any(CobolDocumentModel.class), anyString()))
+        .thenReturn(false);
+    when(SemanticLocations_2.containsToken(any(CobolDocumentModel.class), anyString()))
+        .thenReturn(true);
+    when(semanticLocations_1.definitions(DOCUMENT)).thenReturn(emptyMap());
```

Comment:
Will show a code smell about an unchecked assignment. `Map.of()` should work fine

Reply:
Done